### PR TITLE
Applications: Add launch command for applications addon

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Changelog Description
It is possible to launch application using cli arguments.

## Additional info
Simplest possible approach to launch application using cli.

## Testing notes:
1. Run `./ayon addon applications launch --app maya/2024 --project YourProject --folder /Some/folder/path --task Modeling`.
2. Application should launch.

### CODE REVIEW

File: server_addon/applications/client/ayon_applications/addon.py [hunk 1]
```diff
@@ -171,3 +202,17 @@ def _cli_extract_environments(
 
         with open(output_json_path, "w") as file_stream:
             json.dump(env, file_stream, indent=4)
+
+    def _cli_launch_applications(self, project, folder, task, app):
+        """Produces json file with environment based on project and app.
+
+        Called by farm integration to propagate environment into farm jobs.
```
Comment (BigRoy) [inline server_addon/applications/client/ayon_applications/addon.py:209]: Pretty sure this docstring is incorrect.

File: server_addon/applications/client/ayon_applications/addon.py [hunk 2]
```diff
@@ -134,6 +154,17 @@ def cli(self, addon_click_group):
...
+        (
+            main_group.command(
+                self._cli_launch_applications,
+                name="launch",
+                help="Launch application"
+            )
+            .option("--app", help="Application name")
+            .option("--project", help="Project name")
+            .option("--folder", help="Folder path")
+            .option("--task", help="Task name")
```
Comment (BigRoy) [inline server_addon/applications/client/ayon_applications/addon.py:166]: Should we also add `required=True`?

File: server_addon/applications/client/ayon_applications/addon.py [hunk 3]
```diff
@@ -171,3 +202,17 @@ def _cli_extract_environments(
...
+        Called by farm integration to propagate environment into farm jobs.
+
+        Args:
+            project (str): Project name.
+            folder (str): Folder path.
+            task (str): Task name.
+            app (str): Full application name e.g. 'maya/2024'.
+
+        """
+        self.launch_application(app, project, folder, task,)
```
[the remark this answers is not in the document]
Comment (iLLiCiTiT) [reply server_addon/applications/client/ayon_applications/addon.py]: Nope 🙂

File: server_addon/applications/client/ayon_applications/addon.py (hunk 1 above)
Comment (iLLiCiTiT) [reply server_addon/applications/client/ayon_applications/addon.py]: Copy pasta.

File: server_addon/applications/client/ayon_applications/addon.py (hunk 2 above)
Comment (iLLiCiTiT) [reply server_addon/applications/client/ayon_applications/addon.py:166]: Isn't that by default when `default` is not defined? If not then yes.

Comment (iLLiCiTiT) [reply server_addon/applications/client/ayon_applications/addon.py]: No it was not, added `required=True`.

**Review by MustafaJafar — APPROVED** [review]
This is cool! 
It works on my side.. I used it to launch maya and houdini. 
Also, `--help`  works nicely. 

For reference, Here's an example of how it looks like in terminal on my side.
```powershell
PS E:\Ynput\ayon-launcher> tools/ayon_console.bat --use-dev addon applications launch --app houdini/19-5-805 --project Robo --folder 
"/Assets/Character/warrior_robo" --task cfx  

E:\Ynput\ayon-launcher>goto comment

E:\Ynput\ayon-launcher>cd "E:\Ynput\ayon-launcher\tools\\.."

E:\Ynput\ayon-launcher>.poetry\bin\poetry.exe run python start.py --use-dev addon applications launch --app houdini/19-5-805 --project Robo --folder /Assets/Character/warrior_robo --task cfx
>>> Connected to AYON server http://xxx.xxx.x.x:5000
*** WRN: >>> { AddonsLoader }: [  Addon simple_empty 0.0.1 has no content to import  ]
*** WRN: >>> { AddonsLoader }: [  Addon simple_frontend 0.0.1 has no content to import  ]
*** AYON [1.0.3-dev.1] ------------------------------------------------------------------------------------------------------------  
>>> Using AYON from [ E:\Ynput\ayon-launcher ]
... AYON variant: [ dev (AYON-Dev) ]
... AYON bundle:  [ AYON-Dev ]
>>> loading environments ...
  - global AYON ...
  - for addons ...
>>> [  Setting OCIO environment to config path: C:\Users\Mustafa Taher\AppData\Local\Ynput\AYON\addons\ayon_ocio_1.0.0\ayon_ocio\configs\OpenColorIOConfigs\aces_1.2\config.ocio  ]
>>> [  Setting OCIO_ACTIVE_DISPLAYS environment to: ACES  ]
>>> [  Setting OCIO_ACTIVE_VIEWS environment to: sRGB  ]
>>> [  Launching "houdini/19-5-805" with args (2): ['C:\\Program Files\\Side Effects Software\\Houdini 19.5.805\\bin\\houdinifx.exe', '\\\\storage\\work\\ayon_projects\\Robo\\Assets\\Character\\warrior_robo\\work\\cfx\\robo_warrior_robo_cfx_v228.hip']  ]
```

**Review by BigRoy — APPROVED** [review]
Looks good to me.

**Pull request merged**